### ISSUE TITLE
Support Spring @Async annotation tracing

### ISSUE THREAD

**Issue by jlarroque** [DataDog/dd-trace-java#1510]
Hi team,

I had an endpoint, let's say GET /hey, that was monitored by APM metrics correctly.

In one point, that endpoint called a specific method in my java code. I added @Async annotation to this method in order to start calling it asynchronously.

Since I added that annotation, the entire endpoint is not being monitored anymore, I'm not able to see it again in the APM metrics, and I know that was there before because I have seen it.

Is there anything specific that I should do in order to enable APM metrics when @Async java code is included?

**Comment by devinsba** [comment]
Is this the Spring `@Async` annotation?

**Comment by jlarroque** [comment]
yes

**Comment by richardstartin** [comment]
@jlarroque support for propagating context through `@Async` annotated methods was implemented in #1873 and will be available in the next release

**Comment by yesjinu** [comment]
Is there anyone still struggles with this problem?

I'm calling an async method `B` which triggers sending Firebase push message inside synchronous method `A`.

```
user -> presentation -> call service::method A -> response
                                      ㄴ> call service::method B
```

I can trace method A in the datadog, but method B is omitted. What's the issue?
